src/client/datascience/types.ts

```diff
@@ -314,7 +314,27 @@ export interface IJupyterKernelSpec {
      * Optionally storing the interpreter information in the metadata (helping extension search for kernels that match an interpereter).
      */
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    readonly metadata?: Record<string, any> & { interpreter?: Partial<PythonEnvironment>; originalSpecFile?: string };
+    readonly metadata?: Record<string, any> & {
+        vscode?: {
```

Comment:
Store metadata under `metadata.vscode` (based on jupyter specs, custom metadata must be scoped per application).